pkgs/development/python-modules/mpi4py/default.nix

```diff
@@ -40,20 +41,27 @@ buildPythonPackage rec {
   pythonImportsCheck = [ "mpi4py" ];
 
   nativeCheckInputs = [
-    pytestCheckHook
     mpiCheckPhaseHook
   ];
-  disabledTestPaths = lib.optionals (mpi.pname == "mpich") [
-    # These tests from some reason cause pytest to crash, and therefor it is
-    # hard to debug them. Upstream mentions these tests to raise issues in
-    # https://github.com/mpi4py/mpi4py/issues/418  but the workaround suggested
-    # there (setting MPI4PY_RC_RECV_MPROBE=0) doesn't work.
-    "test/test_util_pool.py"
-    "demo/futures/test_futures.py"
-  ];
 
   __darwinAllowLocalNetworking = true;
 
+  doCheck = true;
+
+  # follow the upstream check process in https://github.com/mpi4py/mpi4py/blob/4.0.3/.github/workflows/ci-test.yml
+  checkPhase = ''
+    runHook preCheck
+
+    for nproc in {1..2}; do
+      echo "Testing mpi4py (np=$nproc)"
+      mpiexec -n $nproc python test/main.py -v
+      echo "Testing mpi4py.futures (np=$nproc)"
+      mpiexec -n $nproc python demo/futures/test_futures.py -v
```

Comment:
What surprises me mostly is the fact that only very specific test files are iterated. Do you have an explanation for that? I also failed to find a solid evidence that upstream does the same.